site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/FhirContextWrapper.java

```diff
@@ -50,8 +51,25 @@ public Bundle executeTrasaction(String uri, Bundle input) {
      * @return the list of resources in the bundle
      */
     public List<IBaseResource> toListOfResources(Bundle bundle) {
-        List<IBaseResource> resp = new ArrayList<>();
-        resp.addAll(BundleUtil.toListOfResources(fhirContext, bundle));
-        return resp;
+        return new ArrayList<>(BundleUtil.toListOfResources(fhirContext, bundle));
     }
+
+    /**
+     * Typed version of to list of resources
+     */
+    public <T extends IBaseResource> List<T> toListOfResourcesOfType(Bundle bundle, Class<T> resourceClass) {
+        return new ArrayList<>(BundleUtil.toListOfResourcesOfType(this.fhirContext, bundle, resourceClass));
+    }
+
+    /**
+     * Return a search instance which can be configured and executed, returning a single type of resource
+     * @param uri FHIR endpoint URI
+     */
+    public IQuery<Bundle> search(final String uri, final String resourceName) {
```

Comment:
Is it possible to also pass in the class type rather than its string name?
(you have a nice example in the method above)

Reply:
Or maybe a class type instead of a string, since the class name (without package, i.e. the short-name) **is** the resource name.